src/leash/pump.py

```diff
@@ -1,118 +1,90 @@
 
-        
+            self.sm.send("M260 A109")
+            self.sm.send("M260 B48")
+            self.sm.send("M260 B27")
+            self.sm.send("M260 S1")
 
-            try:
-                
+            time.sleep(0.03)
 
-                #read addresses 0x06 0x07 and 0x08 for pressure reading
-                self.sm.send("M260 A109 B6 S1")
-                msb = re.search("data:(..)", self.sm.send("M261 A109 B1 S1"))
+            # read addresses 0x06 0x07 and 0x08 for pressure reading
+            self.sm.send("M260 A109 B6 S1")
+            msb = re.search("data:(..)", self.sm.send("M261 A109 B1 S1"))
 
-                self.sm.send("M260 A109 B7 S1")
-                csb = re.search("data:(..)", self.sm.send("M261 A109 B1 S1"))
+            self.sm.send("M260 A109 B7 S1")
+            csb = re.search("data:(..)", self.sm.send("M261 A109 B1 S1"))
 
-                self.sm.send("M260 A109 B8 S1")
-                lsb = re.search("data:(..)", self.sm.send("M261 A109 B1 S1"))
+            self.sm.send("M260 A109 B8 S1")
+            lsb = re.search("data:(..)", self.sm.send("M261 A109 B1 S1"))
 
-                val = msb.group(1)+csb.group(1)+lsb.group(1)
+            val = msb.group(1) + csb.group(1) + lsb.group(1)
 
-                result = int(val, 16)
+            result = int(val, 16)
 
-                if(result & (1 << 23)):
-                    result = result - 2**24
+            if result & (1 << 23):
+                result = result - 2 ** 24
 
-                return result
-            except Exception as e: 
-                print(e)
-                return False
-            
-    def getTemperature(self):
+            return result
+        except Exception as e:
+            log.error(e)
+            return False
 
+    def get_temperature(self):
         try:
             if self.index == "LEFT":
-                #selects vac 1 through multiplexer
+                # selects vac 1 through multiplexer
                 self.sm.send("M260 A112 B1 S1")
             elif self.index == "RIGHT":
                 self.sm.send("M260 A112 B2 S1")
 
             # Assuming sensor is an object or interface to communicate with the sensor
-            
+
             # Read REG0x09 and REG0x0A
             self.sm.send("M260 A109 B9 S1")
-            REG0x09 = re.search("data:(..)", self.sm.send("M261 A109 B1 S1"))
+            reg0x09 = re.search("data:(..)", self.sm.send("M261 A109 B1 S1"))
 
             self.sm.send("M260 A109 B10 S1")
-            REG0x0A = re.search("data:(..)", self.sm.send("M261 A109 B1 S1"))
-            
+            reg0x0_a = re.search("data:(..)", self.sm.send("M261 A109 B1 S1"))
+
             # Calculate the temperature ADC value
-            N = REG0x09 * 256 + REG0x0A
-            
+            n = float(reg0x09.group() * 256 + reg0x0_a.group())
+
             # Determine if temperature is positive or negative
-            if N < 2**15:
+            if n < 2 ** 15:
```

Comment:
`2**15` could be rewritten as `(1 << 15)`, signifies to reader much more about how we are checking 15th bit. And actually, that doesn't make a lot of sense. 2 bytes used, so 16 bits, wherein 16th bit is sign bit, so I think even the original code might be wrong here.